server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyToResponseBody.scala

```diff
@@ -40,7 +46,32 @@ class NettyToResponseBody extends ToResponseBody[NettyResponse, NoStreams] {
         val fileRange = v.asInstanceOf[FileRange]
         (ctx: ChannelHandlerContext) => ChunkedFileNettyResponseContent(ctx.newPromise(), wrap(fileRange))
 
-      case _: RawBodyType.MultipartBody => throw new UnsupportedOperationException
+      case m: RawBodyType.MultipartBody =>
+        val buffers: List[ByteBuf] = v
+          .asInstanceOf[List[RawPart]]
+          .flatMap(part =>
+            m.partType(part.name)
+              .map(bodyType => convertToBuffs(bodyType, part))
+          )
+        (ctx: ChannelHandlerContext) => ByteBufNettyResponseContent(ctx.newPromise(), Unpooled.wrappedBuffer(buffers: _*))
+    }
+  }
+
+  private def convertToBuffs(bodyType: RawBodyType[_], part: Part[Any]): ByteBuf = {
+    bodyType match {
+      case RawBodyType.StringBody(_) =>
+        toPart(part.body, part.contentType, part.name, None)
+      case RawBodyType.ByteArrayBody =>
+        toPart(part.body, part.contentType, part.name, None)
+      case RawBodyType.ByteBufferBody =>
+        toPart(part.body, part.contentType, part.name, None)
+      case RawBodyType.InputStreamBody =>
+        toPart(part.body, part.contentType, part.name, None)
+      case RawBodyType.FileBody =>
+        val fileRange = part.body.asInstanceOf[FileRange]
+        toPart(Files.readString(fileRange.file.toPath), part.contentType, part.name, Some(fileRange.file.getName))
```

Comment:
again, this reads the whole file content into memory - can we provide the data to netty in another way? @rafalambrozewicz you did sth similar, maybe you'll be able to help :)

Reply:
I am not sure if it is possible to do in other way but any help would be welcome.

Reply:
We've checked whether netty has some support for multipart out of the box, but it looks like it doesn't. Not reading everything into memory forces usage of `Chunked` class  i.e. `io.netty.handler.streamChunkedStream`, but turning all parts of multipart into such entity seems not trivial or simply not possible :-|